app/assets/javascripts/components/PipelineSampleReport.jsx

```diff
@@ -3,358 +3,176 @@ class PipelineSampleReport extends React.Component {
-    this.all_categories = props.all_categories || [];
-    this.checked_categories = props.checked_categories || props.all_categories
-    this.pipeline_output_id = props.report_details.pipeline_info.id;
-    this.genus_info = props.genus_info
-    this.sample_id = props.sample_id
+    this.real_length = props.taxonomy_details[0];
+    this.taxonomy_details = props.taxonomy_details[1];
+    this.all_categories = props.all_categories;
+    this.applyViewLevel = this.applyViewLevel.bind(this);
+    this.applyNewFilterThresholds = this.applyNewFilterThresholds.bind(this);
+  }
 
-    this.view_level = ReportFilter.getFilter('view_level') || 'species';
-    this.highest_tax_counts = props.highest_tax_counts;
-    this.defaultSortBy = 'highest_species_nt_zscore';
-    const current_sort = PipelineSampleReport.currentSort();
-    this.state = {
-      sort_query: current_sort.sort_query
-        ? current_sort.sort_query  : `sort_by=${this.defaultSortBy}`
-    };
-    this.applySort = this.applySort.bind(this);
-    this.columnSorting = this.columnSorting.bind(this);
+  refreshPage(overrides) {
+    new_params = Object.assign({}, this.props.report_page_params, overrides);
+    window.location = location.protocol + '//' + location.host + location.pathname + '?' + jQuery.param(new_params);
   }
 
-  static uppCaseFirst(name) {
-    return (name)? name.charAt(0).toUpperCase() + name.slice(1) : name;
+  applyViewLevel(view_level) {
+    this.refreshPage({view_level});
   }
 
-  static currentSort() {
-   const sort_by = ReportFilter.getFilter('sort_by');
-   let current_sort = {};
-   if(sort_by) {
-     current_sort = {
-       sort_query: `sort_by=${sort_by}`
-     }
-   }
-   return current_sort;
+  // applySort needs to be bound at time of use, not in constructor above
+  applySort(sort_by) {
+    this.refreshPage({sort_by});
   }
 
-  applySort(sort_query) {
-    this.setState({ sort_query });
-    const url = PipelineSampleReport.deleteUrlParam(window.location.href, 'sort_by');
-    window.location = (PipelineSampleReport.hasQuery(url))
-      ? `${url}&${sort_query}` : `${url}?${sort_query}`;
+  applyNewFilterThresholds(new_filter_thresholds) {
+    this.refreshPage(new_filter_thresholds);
   }
 
-  static deleteUrlParam(url, parameter) {
-    const queryString = url.split('?');
-    if (queryString.length >= 2) {
-      const prefix = encodeURIComponent(parameter)+'=';
-      const pars = queryString[1].split(/[&;]/g);
-      for (let i = pars.length; i--; i > 0) {
-        if (pars[i].lastIndexOf(prefix, 0) !== -1) {
-          pars.splice(i, 1);
-        }
-      }
-      url = queryString[0] + (pars.length > 0 ? '?' + pars.join('&') : '');
-      return url;
+  render_name(tax_info) {
+    foo = <i>{tax_info.name}</i>;
+    if (tax_info.tax_id > 0) {
+      foo = (
+        <span className="link">
+          <a href={`https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=${tax_info.tax_id}`}>
```

Comment:
<a href={`/pipeline_outputs/${this.pipeline_output_id}/fasta/tax_info.tax_level/${tax_info.tax_id}/NT_or_NR}`>
(or wherever you can get pipeline_output_id from)

Reply:
That's very helpful, thank you, Charles.